[RHELC-724] Move adding of NullHandler to the root logger.

abadger opened the pull request:
@r0x0d added initialize.py for steps that need to be taken early to initialize various subsystems.  Adding of the NullHandler to the root logger (to avoid some libraries initializing themselves in a way that ends up outputting messages multiple times) needs to be done very early. So moving it to initialize.py is the logical choice.

<!-- Write a description of what the PR solves and how -->

<!-- Link to relevant Jira issue, add multiple if necessary -->
Jira Issue: [RHELC-724](https://issues.redhat.com/browse/RHELC-724)

Checklist
- [ ] PR meets acceptance criteria specified in the Jira issue
- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title <!-- For a proper sync with Jira -->
- [ ] Code and tests are documented properly
- [X] The commits are squashed to as few commits as possible (without losing data) <!-- The commits can be squashed to 1 commit, but then we might lose data regarding moving something to a new file and then refactoring for example. Hence squash without losing data -->
- [ ] When merged: Jira issue has been updated to `Release Pending`

r0x0d commented:
I think we actually have a jira issue about this

r0x0d commented:
On another note, the piece of code that was removed was related to python 2.6, and since we dropped support for that, that's why that was removed. 

Just making a comment here about that, so we won't forget in the future why some code was gone, since we spotted that in the PR In the first place.

abadger commented on convert2rhel/initialize.py:
```diff
@@ -49,6 +63,9 @@ def run():
     set_locale()
 
+    # Initialize logging to stop duplicate messages.
+    initialize_logging()
```
I think not.  The reason being that we should try to keep the early initialization just for things that really need to be initialized early.  Otherwise, down the road, we might find out that there is something that needs to be initialized before `main.initalize_logger()` and then we'd have to figure out how to do that.  I will rename the function though since you have a good point that the names are very similar so it is confusing why we have two things that initialize logging.

r0x0d replied:
Nice, good point. At first, I thought that this function was the one from main, but then realized that they are almost the same.

abadger commented:
I've renamed the `initialize.initialize_logging()` function to `initialize.disable_root_logger()` to address one part of @r0x0d 's review.  Rodolfo, let me know if that is insufficient.

I also added the comment about why the python-2.6 portion of the code was removed to the commit message (but made no changes to the code in that commit).  Thanks Rodolfo!  That was good feedback to incorporate.

r0x0d reviewed (approved):
Looks good, @abadger! Thanks for this improvement.

r0x0d commented:
To be merged after the v1.2 release. (Or this week, if someone really wants this to be in 1.2)
